src/app/admin/product-form/product-form.component.ts

```diff
@@ -2,29 +2,49 @@ import { CategoryService } from './../../category.service';
 import { ProductService } from './../../product.service';
 import { Category } from './../../models/category';
 import { Product } from './../../models/product';
-import { Router } from '@angular/router';
-import { Observable } from 'rxjs';
-import { Component } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { Observable, Subscription } from 'rxjs';
+import { Component, OnDestroy } from '@angular/core';
 
 @Component({
   selector: 'app-product-form',
   templateUrl: './product-form.component.html'
 })
-export class ProductFormComponent {
+export class ProductFormComponent implements OnDestroy {
   productsObservable: Observable<[Product]>;
   categoriesObservable: Observable<[Category]>;
   product: Product;
+  subscription: Subscription;
+  id: String;
 
   constructor(
     private router: Router,
+    private route: ActivatedRoute,
     private productService: ProductService,
     private categoryService: CategoryService
   ) {
     this.categoriesObservable = categoryService.getCategories();
+    this.id = this.route.snapshot.paramMap.get('id');
+    if (this.id)
+      this.subscription = this.productService
+        .get(this.id)
+        .subscribe(product => (this.product = product));
```

Comment:
this is simple enough for async pipe no?

Reply:
There is not place in the html to put the async pipe?  This is drawing the id from the product.

Reply:
💯